### DISCUSSION TITLE
Endring av label og description på komponenter

### DISCUSSION THREAD

**Discussion by JorgenMaristuen** [navikt/Designsystemet#1932] [Komponenter]
Vi ønsker å gjøre label og description mer enhetlig på tvers av alle komponenter, samt å forbedre tilgjengelighet og lesbarhet samtidig som nødvendig hierarki mellom teksttyper ivaretas.

I dag brukes det i disse tekstene ulik linjeavstand og tekststørrelser, som er noe som bør kunne forenkles. VI ønsker også å gå bort fra bruk av tekststilen _Detail_, som er 14px og dermed ikke UU-vennlig.

Svarer til https://github.com/navikt/aksel/issues/1881 og https://github.com/navikt/aksel/issues/1917

[Lenke til Slack-tråd om description på radio og checkbox](https://nav-it.slack.com/archives/C7NE7A8UF/p1681911002741809?thread_ts=1637067344.026200&cid=C7NE7A8UF)

**Comment by JorgenMaristuen** [comment]
Foreslår disse justeringene, etter testing på alle komponenter som bruker Label.

1. Samme tekststørrelse og linjeavstand på alle nivåer for _både_ Label og Description (Medium: `BodyShort Medium`, Small: `BodyShort Small`).
2. Label er Semibold, farge: `text-default`.
3. Description er Regular, farge: `text-default`.
4. Descripton på nederste nivå (f.eks description til hver enkelt radio) er Regular, farge `text-subtle`. 
5. Spacing under Label(+Description) settes til 8px.
6. Spacing mellom Label og Description settes til 2px.

Nåværende Label og Description ([figma-prototype](https://www.figma.com/proto/gzyd7P6lDzKhoHVRhU3BHp/Endringsforslag-Label-og-Description?page-id=0%3A1&node-id=2-2&mode=design&scaling=min-zoom&starting-point-node-id=2%3A2&show-proto-sidebar=1&hide-ui=1))
Forslag til ny Label og Description ([figma-prototype](https://www.figma.com/proto/gzyd7P6lDzKhoHVRhU3BHp/Endringsforslag-Label-og-Description?page-id=0%3A1&node-id=2-642&mode=design&scaling=min-zoom&starting-point-node-id=2%3A642&show-proto-sidebar=1&hide-ui=1))

**Reply by finiteattention** [reply]
Hvis vi først skal skille mellom label og description ved å ha label som semibold, må vi faktisk ha hjelpetekst i en lysere farge? Ser at det nok blir god kontrast uansett, men lurer på om det er overflødig å skille typografisk på to ulike vis når vi faktisk kunne hatt et høyere kontrastnivå på description?

**Reply by JorgenMaristuen** [reply]
Text Subtle er tenkt brukt for å skille mellom nivået under komponentens hovedlabel og -description. I dag gjelder det RadioGroup, CheckboxGroup og Switch. En lysere tekstfarge vil her brukes for å skille mellom normal tekst (ikke semibold) og description istedenfor å gå ned i tekststørrelse. På small-varianten er nå description teksten 14 px, som er for lite mht UU. Description under hovedlabelen (semibold) vil alltid være normal tekstfarge.
![image](https://user-images.githubusercontent.com/86107432/233963930-d4d5388f-c144-4caa-b871-72e520677531.png)

**Reply by Trent-NAV** [reply]
Jeg er også usikker på Text Subtle på description-text og tenker det samme som Chris her. 🤔

Etter å ha vært på noen SB observasjoner og sett skjermoppsett osv så er det ikke alltid like bra forhold. Tipper variasjonene er enda større på eksterne flater. Om man har Semibold på label kan man ha samme farge på Regular description tenker jeg. Jeg syns forøvrig det også kan vektes med størrelse og at label burde være litt større enn description.

**Reply by Trent-NAV** [reply]
@JorgenMaristuen Jeg vet ikke om jeg oppfatter det du nevner om "nivåer" her rett, jeg tenker i min kontekst på forholdet mellom en label og en description generelt. Hvis det er flere nivåer av description så blir det et dilemma å vekte med størrelse som middel, men jeg har ikke sett at det har vært behov for nivåforskjeller av description. Da snakker jeg IF.

**Reply by JorgenMaristuen** [reply]
@Trent-NAV Utfordringen ved å gå ned i skriftstørrelse er at vi da kommer ned på 14 px på Small-varianten. WCAG krever minst 16px, og i tillegg er fonten vi bruker relativt liten. Text Subtle er AA-compliant. Fargen er dessuten transparent, slik at den vil beholde kontrastnivået til andre lyse bakgrunnsfarger, ikke bare hvit. Det går an å justere Text Subtle mørkere til AAA, men det vil igjen gå på bekostning av typografisk hierarki og slik sett øke den kognitive lasten.

[Gov.uk](http://gov.uk/) gjør for tiden en review av typografien hvor de bl.a. går helt bort fra fontstørrelse 14 px.
https://designnotes.blog.gov.uk/2022/12/12/making-the-gov-uk-frontend-typography-scale-more-accessible/

**Reply by JorgenMaristuen** [reply]
@Trent-NAV På label og description generelt foreslår vi bare å skille disse ved å bruke semibold på Label og normal tekst på description (ikke noe endring av tekstfarge).

Punkt 3 hadde falt ut i [posten](https://github.com/navikt/aksel/discussions/1932#discussioncomment-5686875) her: "Description er Regular". Beklager om det skapte litt forvirring

**Reply by Trent-NAV** [reply]
Ja, kjenner til WCAG størrelsene, på IF brukes 14px Detail fra DS beskjedent. Body/Long/Small og Body/Short/Small er default for body.  Men jeg tenker for økt lesbarhet at en dimmet / mutet description ikke er optimalt, selv om den er AA-compliant. WCAG tar ikke alltid høyde for optisk opplevelse av kontraster og farger. Ingenting i veien for å øke størrelse på Label mtp typografisk hierarki. 

Hvis man har ulike nivåer av description tenker jeg man heller bør se på hvordan man bruker description.

**Reply by Trent-NAV** [reply]
> @Trent-NAV På label og description generelt foreslår vi bare å skille disse ved å bruke semibold på Label og normal tekst på description (ikke noe endring av tekstfarge).
> 
> Punkt 3 hadde falt ut i [posten](https://github.com/navikt/aksel/discussions/1932#discussioncomment-5686875) her: "Description er Regular". Beklager om det skapte litt forvirring

Den er grei!  😀

**Reply by Trent-NAV** [reply]
[I demoen ](https://www.figma.com/proto/gzyd7P6lDzKhoHVRhU3BHp/Endringsforslag-Label-og-Description?page-id=0%3A1&node-id=2-642&mode=design&scaling=min-zoom&starting-point-node-id=2%3A642&show-proto-sidebar=1&hide-ui=1)så vises subtle description

![radio demo](https://user-images.githubusercontent.com/86958516/233974056-9842f1ca-4d0b-4e5e-b488-03357690617b.png)

**Reply by JorgenMaristuen** [reply]
> [I demoen ](https://www.figma.com/proto/gzyd7P6lDzKhoHVRhU3BHp/Endringsforslag-Label-og-Description?page-id=0%3A1&node-id=2-642&mode=design&scaling=min-zoom&starting-point-node-id=2%3A642&show-proto-sidebar=1&hide-ui=1)så vises subtle description

Ja, men kun på nivå 2, der det er behov for description på et enkelt radiovalg eller checkboxvalg. 
Description under labelen "Hva vil du informere NAV om?" har standard tekstfarge (text-default).

**Reply by Trent-NAV** [reply]
Jeg tenker på input description, altså den som er mutet i demoen. Jeg definerte teksten under øverste label som secondary label. Radio group for IF (Small) nå har label-small (16px) for øverste label og detail/default for description eller secondary label. For **button label** har den body/short/small (16px). Hvis man skal følge den samme strategien for button label description så bør vel den da være detail/default den også? Eller er det den du mener også burde være body/short/small (16px), altså likt med button label men vektes ulikt med text-subtle mot semi-bold?

**Reply by JorgenMaristuen** [reply]
Ja, forslaget er å gå bort fra å bruke tekststørrelse _detail_ i Small-komponenter. 
All tekst i Small-komponenter blir 16 px med linjeavstand 20 px. 
All tekst i Medium-komponenter blir 18px med linjeavstand 24 px.
Øverste label blir semibold. "Button label description" blir text-subtle.
